lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/validation/JarValidator.java

```diff
@@ -87,6 +94,62 @@ public JWTClaimsSet validateRequestJwt(SignedJWT signedJWT, String clientId)
         }
     }
 
+    private void validateScope(String clientId, JWTClaimsSet claimsSet)
+            throws JarValidationException {
+        try {
+            var requestedScope = Scope.parse(claimsSet.getStringClaim("scope"));
+
+            if (wrongNumberOfRequiredScopes(requestedScope)
+                    || scopeNotValidForClient(requestedScope, clientId)) {
+                LOGGER.error(
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "Client not allowed to issue a request with this scope")
+                                .with(LOG_SCOPE.getFieldName(), requestedScope.toString())
+                                .with(LOG_CLIENT_ID.getFieldName(), clientId));
+                throw new JarValidationException(
+                        OAuth2Error.INVALID_SCOPE
+                                .setDescription(
+                                        "Client not allowed to issue a request with this scope")
+                                .setHTTPStatusCode(SC_FORBIDDEN));
+            }
+        } catch (ParameterNotFoundException e) {
+            LOGGER.error(
+                    new StringMapMessage()
+                            .with(
+                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                    "Valid scopes for client not found in config")
+                            .with(LOG_CLIENT_ID.getFieldName(), clientId));
+            throw new JarValidationException(
+                    OAuth2Error.INVALID_CLIENT.setDescription(
+                            "Allowed scopes not found for client"));
+        } catch (ParseException e) {
+            LOGGER.error(
+                    new StringMapMessage()
+                            .with(
+                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                    "Could not parse scope from claims set")
+                            .with(LOG_CLIENT_ID.getFieldName(), clientId));
+            throw new JarValidationException(
+                    OAuth2Error.INVALID_SCOPE.setDescription("Missing scope"));
+        }
+    }
+
+    private boolean wrongNumberOfRequiredScopes(Scope requestedScopes) {
+        return 1
+                != requestedScopes.stream()
+                        .filter(scope -> ONE_OF_REQUIRED_SCOPES.contains(scope.getValue()))
+                        .count();
+    }
+
+    private boolean scopeNotValidForClient(Scope requestedScopes, String clientId) {
+        var validClientScopes =
+                Scope.parse(configService.getSsmParameter(CLIENT_VALID_SCOPES, clientId));
+        return !requestedScopes.stream()
+                .allMatch(scope -> validClientScopes.contains(scope.getValue()));
```

Comment:
Fine to leave as-is, but as a minor style thing, I generally try to keep methods 'positive' and include a verb - e.g. `hasRequiredScope` and `hasValidScopesForClient`.

Reply:
No, that's much better. My brain sometimes has a meltdown when dealing with double negatives so will fix it up